src/tests/testClpMap.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_BASE_TESTCLPMAP_H
+#define SQUID_BASE_TESTCLPMAP_H
+
+#include "compat/cppunit.h"
+#include "base/ClpMap.h"
+
+class testClpMap: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(testClpMap);
+    CPPUNIT_TEST(testMemoryCounter);
+    CPPUNIT_TEST(testConstructor);
+    CPPUNIT_TEST(Entries);
+    CPPUNIT_TEST(PutGetDelete);
+    CPPUNIT_TEST(SetMemLimit);
+    CPPUNIT_TEST(testTtlExpiration);
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+    using testMap = ClpMap<std::string, int>;
+
+    // add a standard set of key-values to the map, up to numElems
+    // the keys and values will start at base and count up numElems
+    void addData(testMap &m, int numElems=10, int base=0, testMap::Ttl ttl=10);
```

Comment:
After attempting this, I'll push back against the type change, because either we change both paramters to size_t (which makes little sense, because base is not a size), or we get into type cast annoyance, so I'll leave them as ints, which is okay given the use case.
I'm okay with the other suggestions and I'll implement them